File: src/main/java/rbq/wtf/lycoris/client/wrapper/wrappers/annotation/WrapperClass.java
package rbq.wtf.lycoris.client.wrapper.wrappers.annotation;

import rbq.wtf.lycoris.client.wrapper.wrappers.annotation.repeat.WrapperClasses;

import java.lang.annotation.*;

@Documented
@Target(ElementType.TYPE)
@Retention(RetentionPolicy.RUNTIME)
@Repeatable(WrapperClasses.class)
public @interface WrapperClass {
    String mcpName();
    String targetMap();
}

File: src/main/java/rbq/wtf/lycoris/client/wrapper/wrappers/annotation/WrapperClass.kt
package rbq.wtf.lycoris.client.wrapper.wrappers.annotation

import rbq.wtf.lycoris.client.wrapper.MapEnum
import rbq.wtf.lycoris.client.wrapper.wrappers.annotation.repeat.WrapperClasses

@MustBeDocumented
@Target(AnnotationTarget.ANNOTATION_CLASS, AnnotationTarget.CLASS)
@Retention(AnnotationRetention.RUNTIME)
@JvmRepeatable(
    WrapperClasses::class
)
annotation class WrapperClass(val mcpName: String, val targetMap: MapEnum)